datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -251,6 +251,19 @@ def http(self):
 
         return self._http
 
+    @property
+    def http_remapper(self):
+        """
+        This is for updating default values before the creation of ``self.http``.
+        Every access makes a copy to avoid affecting other instances.
+        """
+        if self.HTTP_CONFIG_REMAPPER is None:
+            self.HTTP_CONFIG_REMAPPER = {}
+        else:
+            self.HTTP_CONFIG_REMAPPER = copy.deepcopy(self.HTTP_CONFIG_REMAPPER)
```

Comment:
why not just return the copy instead of reasigning?

Reply:
It needs to actually change so the http wrapper may use it